cmd/antrea-agent/agent.go

```diff
@@ -180,17 +184,27 @@ func run(o *Options) error {
 
 	var proxier k8sproxy.Provider
 	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		var nodePortAddresses []*net.IPNet
```

Comment:
I feel we should have a separate feature gate to control NodePort by AntreaProxy or not.

Reply:
Yes, I added a new feature gate `AntreaProxyNodePort` to control it. Also, I updated the workflow here. The parsing of NodeAddress will go only if both `AntreaProxyNodePort` and `AntreaProxy` are enabled.